Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -17,296 +17,516 @@
+void CGContextShowGlyphsWithAdvances(CGContextRef context, const CGGlyph* glyphs, CGSize* advances, unsigned count) {
+    UNIMPLEMENTED();
 }
 
 /**
  @Status Interoperable
 */
-void CGContextTranslateCTM(CGContextRef ctx, CGFloat x, CGFloat y) {
-    ctx->Backing()->CGContextTranslateCTM(x, y);
+void CGContextShowGlyphs(CGContextRef context, const CGGlyph* glyphs, unsigned count) {
+    UNIMPLEMENTED();
 }
+#pragma endregion
 
+#pragma region Userspace Coordinate Transformation
 /**
  @Status Interoperable
 */
-void CGContextScaleCTM(CGContextRef ctx, CGFloat sx, CGFloat sy) {
-    ctx->Backing()->CGContextScaleCTM(sx, sy);
+CGAffineTransform CGContextGetCTM(CGContextRef context) {
+    auto& state = context->CurrentGState();
+    return state.transform;
 }
 
 /**
  @Status Interoperable
 */
-void CGContextRotateCTM(CGContextRef ctx, CGFloat angle) {
-    ctx->Backing()->CGContextRotateCTM(angle);
+void CGContextTranslateCTM(CGContextRef context, CGFloat x, CGFloat y) {
+    CGContextConcatCTM(context, CGAffineTransformMakeTranslation(x, y));
 }
 
 /**
  @Status Interoperable
 */
-void CGContextConcatCTM(CGContextRef ctx, CGAffineTransform t) {
-    ctx->Backing()->CGContextConcatCTM(t);
+void CGContextScaleCTM(CGContextRef context, CGFloat sx, CGFloat sy) {
+    CGContextConcatCTM(context, CGAffineTransformMakeScale(sx, sy));
 }
 
 /**
  @Status Interoperable
 */
-void CGContextSetCTM(CGContextRef ctx, CGAffineTransform transform) {
-    ctx->Backing()->CGContextSetCTM(transform);
+void CGContextRotateCTM(CGContextRef context, CGFloat angle) {
+    CGContextConcatCTM(context, CGAffineTransformMakeRotation(angle));
 }
 
 /**
  @Status Interoperable
 */
-void CGContextRelease(CGContextRef ctx) {
-    if (!ctx) {
-        TraceWarning(TAG, L"CGCOntextRelease NULL!");
-        return;
-    }
+void CGContextConcatCTM(CGContextRef context, CGAffineTransform t) {
+    auto& state = context->CurrentGState();
+    state.transform = CGAffineTransformConcat(state.transform, t);
+}
 
-    CFRelease((id)ctx);
+/**
+ @Status Interoperable
+*/
+void CGContextSetCTM(CGContextRef context, CGAffineTransform transform) {
+    auto& state = context->CurrentGState();
+    state.transform = transform;
 }
+#pragma endregion
 
+#pragma region Image Drawing - Operations
+/// TODO(DH): GH#1078 Image Drawing
 /**
  @Status Interoperable
 */
-void CGContextDrawImage(CGContextRef ctx, CGRect rct, CGImageRef img) {
-    if (img == NULL) {
-        TraceWarning(TAG, L"Img == NULL!");
+void CGContextDrawImage(CGContextRef context, CGRect rect, CGImageRef image) {
+    if (!image) {
+        TraceWarning(TAG, L"Img == nullptr!");
         return;
     }
-    if (ctx == NULL) {
-        TraceWarning(TAG, L"CGContextDrawImage: ctx == NULL!");
+    if (!context) {
+        TraceWarning(TAG, L"CGContextDrawImage: context == nullptr!");
         return;
     }
 
-    ctx->Backing()->CGContextDrawImage(rct, img);
+    { UNIMPLEMENTED(); }
```

Comment:
This was a side effect of a global replace, and will be deleted with time.

---
In reply to: [83312670](https://github.com/Microsoft/WinObjC/pull/1126#discussion_r83312670) [](ancestors = 83312670)